unlock-app/src/hooks/useProvider.ts

```diff
@@ -28,7 +28,7 @@ interface Config {
   httpProvider: string
 }
 
-export const useProvider = () => {
+export const useProvider = (providerAdapter?: any) => {
```

Comment:
nit: can we stick to `provider` here?

I think this functionality will be used for more than just the paywall-passed provider. For example I think this is who we will also suppport web3Modal in unlock-app(outside of the pywall context)